moz_telemetry/sandboxes/heka/analysis/moz_telemetry_new_doctype_monitor.lua

```diff
@@ -0,0 +1,141 @@
+
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Mozilla Telemetry New DocType Monitor
+
+Monitors the telemetry stream for new document types and alerts if 
+they meet a volume threshold.
+
+## Sample Configuration
+```lua
+filename = "moz_telemetry_new_doctype_monitor.lua"
+message_matcher = "Type == 'telemetry'"
+ticker_interval = 60
+preserve_data = true
+threshold = 100
+
+alert = {
+  disabled = false,
+  prefix = false,
+  threshold = 90,  -- one alert every 90 minutes
+  modules = {
+    email = {recipients = {"amiyaguchi@mozilla.com"}},
+  }
+}
+```
+--]]
+
+_PRESERVATION_VERSION = 0
+
+local alert = require "heka.alert"
+local table = require "table"
+
+-- Helper function to generate a set
+function Set(list)
+    local set = {}
+    for _, l in ipairs(list) do set[l] = true end
+    return set
+end
+
+-- Maintained in the puppet repository: https://github.com/mozilla-services/puppet-config/blob/master/pipeline/yaml/app/pipeline.yaml#L78
```

Comment:
This is another reason this should be cfg based then it only has to be maintained in one place.

Reply:
A long list in the configuration feels wrong, but looking at the tests, the configuration can just be a file.

Reply:
Yes, the cfg will be a file generated by puppet config